src/clojure/puppetlabs/ssl_utils/simple.clj

```diff
@@ -54,38 +88,34 @@
    have a :keylength for specifying this value."
   4096)
 
-(defn gen-keys
+(schema/defn ^:always-validate gen-keys :- SSLKeyPair
   "Generate public and private keys and the X500 name for the given `certname'.
    An optional map may be provided to specify:
 
    * :keylength  Bit length to use for the public/private keys;
                  defaults to 4096."
   ([certname] (gen-keys certname {}))
-  ([certname options]
-   {:pre [(string? certname)
-          (map? options)]
-    :post [(ssl-keys? %)]}
+  ([certname :- schema/Str
+    options :- SSLOptions]
    (let [keylength (get options :keylength default-keylength)
          keypair (ssl-utils/generate-key-pair keylength)]
      {:public-key (ssl-utils/get-public-key keypair)
       :private-key (ssl-utils/get-private-key keypair)
       :x500-name (ssl-utils/cn certname)
       :certname certname})))
 
-(defn gen-cert*
+(schema/defn ^:always-validate gen-cert* :- X509Certificate
   "Internal helper function to generate a certificate; see `gen-cert' for the
    public version of this function.
    An optional map may be provided to specify:
 
    * :extensions  List of certificate extensions to include on the certificate;
                   defaults to []."
   ([ca-keys host-keys serial] (gen-cert* ca-keys host-keys serial {}))
```

Comment:
k, I suppose that's fine.  Not too much work for people to look at the other arities to know how they want to type the args for the lower arity version that they want to use.